src/key.cpp

```diff
@@ -332,6 +334,40 @@ bool CKey::Derive(CKey& keyChild, ChainCode &ccChild, unsigned int nChild, const
     return ret;
 }
 
+// Returns just the x-coordinate in big endian
+static int bip324_ecdh_hash(unsigned char* output, const unsigned char* x32, const unsigned char* ours, const unsigned char* theirs, void* data)
+{
+    memcpy(output, x32, 32);
+    return 1;
+}
+
+std::optional<ECDHSecret> CKey::ComputeBIP324ECDHSecret(const Span<const std::byte> their_ellswift,
+                                                        const Span<const std::byte> our_ellswift,
+                                                        bool initiating) const
+{
+    uint8_t xonly_ecdh[ECDH_SECRET_SIZE];
```

Comment:
ab46788: uint8_t can be made std::byte.

Reply:
@stratospher Upon further thought after our conversation, I think unsigned char makes sense here as that's exactly what all the functions it is used with require. So I'm going with that.